### ISSUE TITLE
`use_late_for_private_fields_and_variables` should have tests with patterns

### ISSUE THREAD

**Issue by srawlins** [dart-lang/linter#4180]
Child issue of https://github.com/dart-lang/linter/issues/3624

I think there should be a test case with an assignment pattern like:

```dart
int? _i;
m() {
  _i?.abs();
  (_i, ) = (null, );
}
```

in which the lint rule would allow that `_i` can be nullable, as it is assigned `null`, in a pattern.
